// @syncfusion/ej2-angular-navigations/src/accordion/items.directive.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Directive, ViewContainerRef, ContentChildren, ContentChild } from '@angular/core';
import { ComplexBase, ArrayBase, setValue } from '@syncfusion/ej2-angular-base';
import { Template } from '@syncfusion/ej2-angular-base';
var input = ['content', 'cssClass', 'expanded', 'header', 'iconCss'];
var outputs = [];
/**
 * ItemDirective represent a item of the Essential JS 2 Angular Accordion.
 * ```html
 * <ejs-accordion >
 *   <e-accordionitems>
 *    <e-accordionitem header='Header1'></e-accordionitem>
 *    <e-accordionitem header='Header2' content='Content2'></e-accordionitem>
 *   </e-accordionitems>
 * </ejs-accordion>
 * ```
 */
var AccordionItemDirective = /** @class */ (function (_super) {
    __extends(AccordionItemDirective, _super);
    function AccordionItemDirective(viewContainerRef) {
        var _this = _super.call(this) || this;
        _this.viewContainerRef = viewContainerRef;
        setValue('currentInstance', _this, _this.viewContainerRef);
        _this.registerEvents(outputs);
        return _this;
    }
    AccordionItemDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'e-accordionitems>e-accordionitem',
                    inputs: input,
                    outputs: outputs,
                    queries: {}
                },] },
    ];
    /** @nocollapse */
    AccordionItemDirective.ctorParameters = function () { return [
        { type: ViewContainerRef, },
    ]; };
    AccordionItemDirective.propDecorators = {
        'header': [{ type: ContentChild, args: ['header',] },],
        'content': [{ type: ContentChild, args: ['content',] },],
    };
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], AccordionItemDirective.prototype, "header", void 0);
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], AccordionItemDirective.prototype, "content", void 0);
    return AccordionItemDirective;
}(ComplexBase));
export { AccordionItemDirective };
/**
 * AccordionItem Array Directive
 * @private
 */
var AccordionItemsDirective = /** @class */ (function (_super) {
    __extends(AccordionItemsDirective, _super);
    function AccordionItemsDirective() {
        return _super.call(this, 'items') || this;
    }
    AccordionItemsDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'ejs-accordion>e-accordionitems',
                    queries: {
                        children: new ContentChildren(AccordionItemDirective)
                    },
                },] },
    ];
    /** @nocollapse */
    AccordionItemsDirective.ctorParameters = function () { return []; };
    return AccordionItemsDirective;
}(ArrayBase));
export { AccordionItemsDirective };
