// @syncfusion/ej2-angular-navigations/src/toolbar/items.directive.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Directive, ViewContainerRef, ContentChildren, ContentChild } from '@angular/core';
import { ComplexBase, ArrayBase, setValue } from '@syncfusion/ej2-angular-base';
import { Template } from '@syncfusion/ej2-angular-base';
var input = ['align', 'cssClass', 'htmlAttributes', 'id', 'overflow', 'prefixIcon', 'showAlwaysInPopup', 'showTextOn', 'suffixIcon', 'template', 'text', 'tooltipText', 'type', 'width'];
var outputs = ['click'];
/**
 * ItemDirective represent a item of the Essential JS 2 Angular Toolbar.
 * ```html
 * <ejs-toolbar >
 *   <e-items>
 *    <e-item text='Cut'></e-item>
 *    <e-item text='Copy'></e-item>
 *   </e-items>
 * </ejs-toolbar>
 * ```
 */
var ItemDirective = /** @class */ (function (_super) {
    __extends(ItemDirective, _super);
    function ItemDirective(viewContainerRef) {
        var _this = _super.call(this) || this;
        _this.viewContainerRef = viewContainerRef;
        setValue('currentInstance', _this, _this.viewContainerRef);
        _this.registerEvents(outputs);
        return _this;
    }
    ItemDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'e-items>e-item',
                    inputs: input,
                    outputs: outputs,
                    queries: {}
                },] },
    ];
    /** @nocollapse */
    ItemDirective.ctorParameters = function () { return [
        { type: ViewContainerRef, },
    ]; };
    ItemDirective.propDecorators = {
        'template': [{ type: ContentChild, args: ['template',] },],
    };
    __decorate([
        Template(),
        __metadata("design:type", Object)
    ], ItemDirective.prototype, "template", void 0);
    return ItemDirective;
}(ComplexBase));
export { ItemDirective };
/**
 * Item Array Directive
 * @private
 */
var ItemsDirective = /** @class */ (function (_super) {
    __extends(ItemsDirective, _super);
    function ItemsDirective() {
        return _super.call(this, 'items') || this;
    }
    ItemsDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'ejs-toolbar>e-items',
                    queries: {
                        children: new ContentChildren(ItemDirective)
                    },
                },] },
    ];
    /** @nocollapse */
    ItemsDirective.ctorParameters = function () { return []; };
    return ItemsDirective;
}(ArrayBase));
export { ItemsDirective };
