// @syncfusion/ej2-angular-navigations/src/menu/items.directive.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { Directive, ViewContainerRef, ContentChildren } from '@angular/core';
import { ComplexBase, ArrayBase, setValue } from '@syncfusion/ej2-angular-base';
var input = ['iconCss', 'id', 'items', 'separator', 'text', 'url'];
var outputs = [];
var MenuItemDirective = /** @class */ (function (_super) {
    __extends(MenuItemDirective, _super);
    function MenuItemDirective(viewContainerRef) {
        var _this = _super.call(this) || this;
        _this.viewContainerRef = viewContainerRef;
        setValue('currentInstance', _this, _this.viewContainerRef);
        _this.registerEvents(outputs);
        return _this;
    }
    MenuItemDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'ejs-menu>e-menu-items>e-menu-item>',
                    inputs: input,
                    outputs: outputs,
                    queries: {}
                },] },
    ];
    /** @nocollapse */
    MenuItemDirective.ctorParameters = function () { return [
        { type: ViewContainerRef, },
    ]; };
    return MenuItemDirective;
}(ComplexBase));
export { MenuItemDirective };
/**
 * MenuItem Array Directive
 * @private
 */
var MenuItemsDirective = /** @class */ (function (_super) {
    __extends(MenuItemsDirective, _super);
    function MenuItemsDirective() {
        return _super.call(this, 'items') || this;
    }
    MenuItemsDirective.decorators = [
        { type: Directive, args: [{
                    selector: 'ejs-menu>e-menu-items',
                    queries: {
                        children: new ContentChildren(MenuItemDirective)
                    },
                },] },
    ];
    /** @nocollapse */
    MenuItemsDirective.ctorParameters = function () { return []; };
    return MenuItemsDirective;
}(ArrayBase));
export { MenuItemsDirective };
